### PR TITLE
[CI/CD] [Ansible] Update E2E testing pipeline, Ansible fixes

### PR DESCRIPTION
**Pull request by ionutbalutoiu**
Jenkins CI/CD updates:
* Move packer VHD templates to their own directory.
* Add `build_azure_managed_images.Jenkinsfile` pipeline to
  generate Azure managed images for the Jenkins CI/CD
  environment.
* Integrate the `custom_label.Jenkinsfile` pipeline into
  the main `Jenkinsfile` pipeline.
* Update E2E testing pipeline.
* Fixes Docker images build.
* Remove unused Jenkins CI/CD pipelines
* Use explicit Docker tag when caching images.
  This is used when caching the Docker images on the Ubuntu
  agents used within the Jenkins environment.

Ansible updates:
* Bump expected GCC version for Ubuntu 18.04.
* Use `playbook_dir` default Ansible variable, and relative
  paths to it, instead of the exec call to `git` to find
  out the source root directory.
* Replace `dcap_testing_node` parameter with
  `launch_configuration` when configuring Windows machines.
* Add retries to CMake download.

Scripts updates:
* Allow long file paths on the Windows machines configured
  with `install-windows-prereqs.ps1` script. The Open Enclave
  source directory tree might have file paths exceeding the
  default limit of 260 characters (especially the 3rd party
  libraries file paths). Unless the git directory location is
  short (for example `C:\` or `D:\`), there is a high chance
  that file paths will exceed 260 characters, leading to
  `Filename too long` file system erros. The fix for this is
  to disable the file path limit via the proper registry key.

Co-authored-by: Marius Oprin <moprin@cloudbasesolutions.com>
Signed-off-by: Ionut Balutoiu <ibalutoiu@cloudbasesolutions.com>

### CODE REVIEW

File: scripts/install-windows-prereqs.ps1 [hunk 1]
```diff
@@ -662,6 +662,10 @@ try {
     Install-Node
     Install-VCRuntime
 
+    Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem" `
```
Comment (BRMcLaren) [inline scripts/install-windows-prereqs.ps1]: Could you add a comment detailing that this is for the long path issue encountered in ci? It's not obvious, unfortunately.

File: scripts/ansible/roles/windows/az-dcap-client/tasks/validation.yml [hunk 2]
```diff
@@ -19,36 +19,37 @@
   register: file
   failed_when: file.stat.isreg == False
 
-- name: Azure DCAP Client | Check for the LC driver registry key
```
Comment (BRMcLaren) [inline scripts/ansible/roles/windows/az-dcap-client/tasks/validation.yml]: Was this file intentionally modified? It looks to be white space changes without any logic, if so, could we revert.

**Comment by BRMcLaren** [comment]
bors try

File: scripts/ansible/roles/linux/openenclave/tasks/environment-setup.yml [hunk 3]
```diff
@@ -24,6 +24,10 @@
     src: https://cmake.org/files/v3.13/cmake-3.13.1-Linux-x86_64.tar.gz
     dest: "{{ cmake_prefix }}"
     remote_src: yes
+  retries: 10
```
[the remark this answers is not in the document]
Comment (tehsiuhuang) [reply scripts/ansible/roles/linux/openenclave/tasks/environment-setup.yml]: This is what we are looking for. Thanks!

File: scripts/install-windows-prereqs.ps1 (hunk 1 above)
Comment (ionutbalutoiu) [reply scripts/install-windows-prereqs.ps1]: Yes. I'll do this.

File: scripts/ansible/roles/windows/az-dcap-client/tasks/validation.yml (hunk 2 above)
Comment (ionutbalutoiu) [reply scripts/ansible/roles/windows/az-dcap-client/tasks/validation.yml]: Yes. These validation tasks only need to run in this case:
```
when: launch_configuration is defined and launch_configuration == 'SGX1FLC'
```

The Ansible `launch_configuration` parameter was added in this pull request for a clear separation between Windows machines configured:
* Without SGX driver installed: `SGX1-NoDriver` or `SGX1FLC-NoDriver`. Windows machines could either have driver installed via Windows Update or they could simply lack SGX support.
* With SGX support (the driver is installed): `SGX1`.
* With SGX1FLC support (the driver is installed): `SGX1FLC`.

**Comment by ionutbalutoiu** [comment]
> Please squash the commits, thanks!

Fixed

**Comment by ionutbalutoiu** [comment]
@BRMcLaren @mingweishih 

Addressed all the comments, feel free to merge the pull request if everything looks good now.

File: scripts/ansible/roles/windows/az-dcap-client/tasks/validation.yml (hunk 2 above)
Comment (BRMcLaren) [reply scripts/ansible/roles/windows/az-dcap-client/tasks/validation.yml]: Oh ok, that makes sense. Thanks for explaining

**Pull request merged**